File: ProjectStart/ProjectStart.WebApp/ClientApp/Shared/AppModels/FilterQueryViewModel.ts

export interface IFilterQueryViewModel  {
    q: { [key: string]: any; };
    page: number;
    pageSize: number;
    withTotalRecord: boolean;
}
export class FilterQueryViewModel  implements IFilterQueryViewModel {
    q: { [key: string]: any; };
    page: number;
    pageSize: number;
    withTotalRecord: boolean;
}

File: ProjectStart/ProjectStart.Common/ViewModel/FilterQueryViewModel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace ProjectStart.Common.ViewModel
{
    public class FilterQueryViewModel
    {
        public IDictionary<string,object> Q { get; set; }
        public int Page { get; set; }
        public int PageSize { get; set; }
        [JsonIgnore]
        public bool WithTotalRecord { get; set; }
    }
}
